pkg/operation/botanist/addons.go

```diff
@@ -443,8 +409,105 @@ func (b *Botanist) generateCoreAddonsChart() (*chartrenderer.RenderedChart, erro
+			},
+			"kubernetesVersion": b.Shoot.Info.Spec.Kubernetes.Version,
+		}
+		vpnTLSAuthSecret = b.Secrets["vpn-seed-tlsauth"]
+		vpnShootConfig   = map[string]interface{}{
+			"podNetwork":     b.Shoot.Networks.Pods.String(),
+			"serviceNetwork": b.Shoot.Networks.Services.String(),
+			"tlsAuth":        vpnTLSAuthSecret.Data["vpn.tlsauth"],
+			"podAnnotations": map[string]interface{}{
+				"checksum/secret-vpn-shoot": b.CheckSums["vpn-shoot"],
+			},
+			"kubernetesVersion": b.Shoot.Info.Spec.Kubernetes.Version,
+		}
+		konnectivityAgentSecret = b.Secrets["konnectivity-agent"]
+		vpnShootSecret          = b.Secrets["vpn-shoot"]
+
+		shootClient = b.K8sShootClient.Client()
+	)
+
+	// Konnectivity agent related values
+	konnectivityAgent, err := b.InjectShootShootImages(konnectivityAgentConfig, common.KonnectivityAgentImageName)
+	if err != nil {
+		return nil, err
+	}
+	values["konnectivity-agent"] = common.GenerateAddonConfig(konnectivityAgent, true)
+
+	// OpenVPN related values
+	if openvpnDiffieHellmanSecret, ok := b.Secrets[common.GardenRoleOpenVPNDiffieHellman]; ok {
+		vpnShootConfig["diffieHellmanKey"] = openvpnDiffieHellmanSecret.Data["dh2048.pem"]
+	}
+
+	if v := b.Shoot.GetNodeNetwork(); v != nil {
+		vpnShootConfig["nodeNetwork"] = *v
+		shootInfo["nodeNetwork"] = *v
+	}
+
+	vpnShoot, err := b.InjectShootShootImages(vpnShootConfig, common.VPNShootImageName)
+	if err != nil {
+		return nil, err
+	}
+	values["vpn-shoot"] = common.GenerateAddonConfig(vpnShoot, true)
+
+	if kubernetesVersionGeq118 {
+		tunnelSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "konnectivity-agent",
+				Namespace: metav1.NamespaceSystem,
+			},
+			Type: corev1.SecretTypeOpaque,
+			Data: konnectivityAgentSecret.Data,
+		}
+
+		// TODO: remove when all Gardener supported versions are >= 1.18
+		// Remove vpn-shoot leftovers before switching to konnectivity tunnels
+		secret, err := controller.GetSecretByReference(context.TODO(), shootClient, &corev1.SecretReference{Namespace: metav1.NamespaceSystem, Name: "vpn-shoot"})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+
+		if secret != nil {
+			if err := b.ChartApplierShoot.Delete(context.TODO(), filepath.Join(common.ChartPath, "shoot-core", "components", "charts", "vpn-shoot"), "shoot-core", metav1.NamespaceSystem); err != nil {
+				return nil, err
+			}
+
+			// Delete the vpn-shoot secret wich was created outside of the chart scope.
+			if err := shootClient.Delete(context.TODO(), secret); err != nil {
+				return nil, err
+			}
+		}
+	} else {
+		tunnelSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "vpn-shoot",
+				Namespace: metav1.NamespaceSystem,
+			},
+			Type: corev1.SecretTypeOpaque,
+			Data: vpnShootSecret.Data,
+		}
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(context.TODO(), shootClient, tunnelSecret, func() error { return nil }); err != nil {
```

Comment:
Why is the tunnel secret not created as part of the Helm chart? This ensures that it's reconciled by GRM.

Reply:
I thought we need to fetch the data in a separate step from our secrets after it got applied and update it, so decided to do it like the vpn-shoot secret.

Reply:
Hm, can you explain more? Which data needs to be fetched in a separate step?
Also, what do you mean by "like the vpn-shoot secret"?

Reply:
> Which data needs to be fetched in a separate step?
Ignore my first comment, I mixed something up. 

> what do you mean by "like the vpn-shoot secret"?

I meant this https://github.com/gardener/gardener/blob/de96851b72d7956e84afc1e5605f75b4f6b01fdd/pkg/operation/botanist/addons.go#L421 thought we already create the VPN shoot secret separately, so using the same call did it for the konnectivity-agent.

Reply:
Hm, I have no idea why this exists. I see I did it 9 months ago as part of extensibility, but I don't recall anymore why it was needed. Probably this can be deleted.

Reply:
I added the secret to the chart, its needed for the vpn-deployment.